thirdparty/kyaml/runfn/runfn.go

```diff
@@ -110,6 +108,53 @@ func (r RunFns) Execute() error {
 	return r.runFunctions(nodes, output, fltrs)
 }
 
+func (r RunFns) getPipelineConfigFilterFn() (kio.LocalPackageSkipFileFunc, error) {
+	fnConfigPaths := sets.String{}
+	// TODO: eventually walking all sub-packages should be handled by
+	// pkg.FunctionConfigFilePaths
+	err := filepath.Walk(r.absPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed to traverse package: %w", err)
+		}
+		if !info.IsDir() {
+			return nil
+		}
+		p, err := pkg.New(path)
+		if err != nil {
+			// we ignore the error here and consider this is not
+			// a package but don't return filepath.SkipDir because
+			// there might be sub-package in the sub-directory
+			return nil
+		}
+		pl, err := p.Pipeline()
+		if err != nil {
+			return nil
+		}
+		paths := pkg.FunctionConfigFilePaths(pl).List()
+		for i := range paths {
+			paths[i] = filepath.Join(path, paths[i])
+			// convert to relative path to top level package
+			paths[i], err = filepath.Rel(r.absPath, paths[i])
```

Comment:
writing a new function (in context of previous comment) because of this specific logic on translating paths make more sense.

Reply:
done